admin/spec/service/data-access/pupil.data.service.spec.js

```diff
@@ -66,4 +68,29 @@ describe('pupil.data.service', () => {
       expect(typeof service.update).toBe('function')
     })
   })
+
+  describe('#getPupils', () => {
+    beforeEach(() => {
+      const pupil1 = pupilMock
+      const pupil2 = pupilMock
+      pupil2.id = '595cd5416e5ca13e48ed2520'
+      pupil2.pin = 'f55sg'
+      sandbox.mock(School).expects('findOne').chain('lean').chain('exec').returns(schoolMock)
+      sandbox.mock(Pupil).expects('getPupils').chain('exec').returns([ pupil1, pupil2 ])
+      service = proxyquire('../../../services/data-access/pupil.data.service', {
+        '../../models/pupil': Pupil,
+        '../../models/school': School
+      })
+    })
+
+    it('has a get Pupils method', () => {
+      expect(typeof service.getPupils).toBe('function')
+    })
+
+    it('returns a school data object and a list of pupils', async () => {
+      const data = await pupilDataService.getPupils(schoolMock._id)
+      expect(data.schoolData._id).toBe(9991001)
+      expect(data.pupils.length).toBe(2)
+    })
+  })
```

Comment:
what value is there in testing a data access component, if all it does is execute a query?

can we not rely on the e2e tests for that?

Reply:
@GeorgeChatzigiannis @GuyHarwood : I believe it would be useful to check that the mock's behaviour was called: 
e.g. `pupilMock.verify()`  where pupilMock is the mock returned from `sandbox.mock(Pupil)`.  ditto for School.